### DISCUSSION TITLE
Combining datasets

### DISCUSSION THREAD

**Discussion by asarnow** [nextpyp/.github#88] [Q&A] [answered]
Sorry if this was discussed, I couldn't find it. Is there any way to combine datasets? For example, I was able to take a set the same grids to the microscope several times, and each collection run has its own gain reference. I also measured or used slightly different electron doses.

For a few I may even have used a different magnification or collected in super-res, but it's understandable combining these is more complicated.

Also related - I have datasets with inconsistent fiducials, and certain TS are best aligned with patches while others work best with fiducials, and it would be nice to be able to just reprocess those needing fiducial alignment for example and then use them all together after that.

**Comment by abartesaghi** [comment, marked as answer]
Merging datasets with different gain references is not currently supported. However, you can process the datasets independently and then create a new folder with links to the gain-corrected tilt-series located in the `mrc/` folder of each project. You will then need to create a new project and import the data from the folder containing the symlinks to the tilt-series from both datasets. The only step you won't be able to execute on the merged dataset is the tilt-frame processing because that will require using the two different gain references. This strategy assumes that the datasets you want to merge have the same magnification. If one dataset was collected in super resolution mode and the other not, you'll need to make sure that the gain-corrected super-resolution tilt-series are first binned 2x before merging the datasets.

You don't need to worry about the dose parameters because nextPYP will do the dose weighting during 3D refinement based on the data itself. The dose parameters are only needed for doing the tilt-series weighting through `unblur` or `motioncor3` (prior to tomogram reconstruction), but this does not affect any 3D refinement operations downstream.

Regarding your last point, the default behavior in nextPYP is to first try alignment with gold fiducials. If that fails, nextPYP will automatically fall back to patch based alignment using IMOD. Other than that, there is no way to align tilt series with different methods within the same project. You could, however, use the strategy above to merge two subsets of the data that were aligned using different strategies.
